link/src/main/java/com/stripe/android/link/ui/wallet/WalletScreen.kt

```diff
@@ -0,0 +1,390 @@
+                shape = MaterialTheme.shapes.medium,
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = MaterialTheme.colors.secondary
+                )
+            ) {
+                Text(
+                    text = stringResource(id = R.string.wallet_pay_another_way),
+                    color = MaterialTheme.colors.onSecondary
+                )
+            }
+        }
+    }
+}
+
+@Composable
+internal fun CollapsedPaymentDetails(
+    selectedPaymentMethod: ConsumerPaymentDetails.PaymentDetails,
+    onClick: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(64.dp)
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.colors.onBackground,
+                shape = MaterialTheme.shapes.large
+            )
+            .clickable(onClick = onClick),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Text(
+            text = stringResource(id = R.string.wallet_pay_with),
+            modifier = Modifier.padding(horizontal = 20.dp),
+            color = MaterialTheme.colors.onBackground
+        )
+        if (selectedPaymentMethod is ConsumerPaymentDetails.Card) {
+            CardDetails(card = selectedPaymentMethod)
+        }
+        Spacer(modifier = Modifier.weight(1f))
+        Icon(
+            painter = painterResource(id = R.drawable.ic_link_chevron),
+            contentDescription = stringResource(id = R.string.wallet_expand_accessibility),
+            modifier = Modifier.padding(horizontal = 20.dp),
+            tint = MaterialTheme.colors.onBackground
+        )
+    }
+}
+
+@Composable
+internal fun ExpandedPaymentDetails(
+    paymentDetails: List<ConsumerPaymentDetails.PaymentDetails>,
+    selectedIndex: Int,
+    onIndexSelected: (Int) -> Unit,
+    onAddNewPaymentMethodClick: () -> Unit,
+    onCollapse: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.colors.onBackground,
+                shape = MaterialTheme.shapes.large
+            )
+    ) {
+        Row(
+            modifier = Modifier
+                .height(44.dp)
+                .padding(start = 20.dp, top = 20.dp, end = 20.dp)
+                .clickable(onClick = onCollapse),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(
+                text = stringResource(id = R.string.wallet_expanded_title),
+                color = MaterialTheme.colors.onPrimary,
+                style = MaterialTheme.typography.button
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Icon(
+                painter = painterResource(id = R.drawable.ic_link_chevron),
+                contentDescription = stringResource(id = R.string.wallet_expand_accessibility),
+                modifier = Modifier.rotate(180f),
```

Comment:
Is it performant to rotate this chevron? Would it be better to just have a different resource for it?